pkg/terraform.go

```diff
@@ -233,6 +234,20 @@ func (e *Executor) processTfPlan(repo Repo, dryRun bool, envVars map[string]stri
 	var output map[string]tfexec.OutputMeta
 
 	if dryRun {
+		// Provider validation aims to prevent common errors like missing variables or not declaring the S3 backend
+		// it only runs in dry run mode since app-interface gates the apply step
+		log.Printf("Validating terraform provider configuration for %s", repo.Name)
+		validationResult := ValidateTerraformRepo(dir, GetDefaultValidationConfig())
+		if !validationResult.Valid {
+			var errorMsg strings.Builder
+			errorMsg.WriteString("Terraform provider validation failed:")
+			for _, err := range validationResult.Errors {
+				errorMsg.WriteString(fmt.Sprintf("\n  - %s", err.Error()))
+			}
+			return nil, fmt.Errorf("terraform provider validation failed: %s", errorMsg.String())
+		}
+		log.Printf("Terraform provider validation passed for %s", repo.Name)
+
```

Comment:
we only need to check if backend is set as `s3`, right? why not inspect `terraform.tfstate` to check `.backend.type` directly?

Reply:
I moved the validation code to before the init step happens. IMO its best if we can avoid initializing before the backend is setup to prevent tenants from creating misconfigurations in AWS and also totally prevent a local backend setup from being created.